Microsoft.Research/ClousotMain/Caching/CacheManager.cs

```diff
@@ -163,65 +162,21 @@ public int NbCacheHits
                     Console.WriteLine("[cache] Slice time stamp : {0}", this.sliceTime);
                 }
 
-                this.cacheAccessor = CreateCacheAccessorAsync(cacheAccessorFactories, options).Result;
-            }
-            finally
-            {
-                this.StopTimer(Timer.ctor);
-            }
-        }
-
-        /// <summary>
-        /// Creates <see cref="IClousotCache"/> in parallel and returns the first one that succeeds.
-        /// </summary>
-        /// <remarks>
-        /// Resulting task will never fail and if the method would not be able to connect
-        /// to the cache, result of the task would be null. 
-        /// Implementation could be refined by adding cancellation.
-        /// </remarks>
-        private async Task<IClousotCache> CreateCacheAccessorAsync(
-            IEnumerable<IClousotCacheFactory> cacheAccessorFactories, GeneralOptions options)
-        {
-            var factories = cacheAccessorFactories.ToList();
-
-            List<Task<IClousotCache>> factoryTasks = 
-                factories.Select(factory => factory.CreateAndCheckAsync(options)).ToList();
-
-            ObservePotentialExceptions(factoryTasks);
-
-            while (factoryTasks.Count > 0)
-            {
-                Task<IClousotCache> factoryTask = 
-                    await Task.WhenAny(factoryTasks).ConfigureAwait(false);
-
-                if (factoryTask.Status == TaskStatus.RanToCompletion && factoryTask.Result != null)
+                foreach (var factory in cacheAccessorFactories)
```

Comment:
I think even in this case we still would benefit from extracting this logic out of the constructor itself.